hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -437,12 +438,27 @@ public OzoneVolume getVolumeDetails(String volumeName)
   }
 
   @Override
-  public OzoneVolume getS3VolumeDetails() throws IOException {
-    OmVolumeArgs volume = ozoneManagerClient.getS3Volume();
+  public OzoneVolume getS3Volume() throws IOException {
+    final GetS3VolumeResponse resp = getS3VolumeInfo();
+
+    // Update user principal, to be used for KMS
+    S3Auth s3Auth = getThreadLocalS3Auth();
+    if (s3Auth != null) {
+      LOG.info("Updating S3Auth->UserPrincipal");  // TODO: Switch to debug
```

Comment:
We can add some comment here to indicate why are using principal and not accessID here.

Reply:
done